R/bcdc_search.R

```diff
@@ -93,6 +93,41 @@ bcdc_list_group_records <- function(group) {
 
 }
 
+#' @export
+#' @describeIn bcdc_list_organization_records
+#'
+bcdc_list_organizations <- function() bcdc_search_facets("organization")
+
+#' Retrieve organization information for B.C. Data Catalogue
+#'
+#' Returns a tibble of organizations or records. Organizations can be viewed here:
+#' https://catalogue.data.gov.bc.ca/organizations or accessed directly from R using `bcdc_list_organizations`
+#'
+#' @param organization Name of the organization
+#' @export
+#' @examples
+#' \donttest{
+#' try(
+#'   bcdc_list_organization_records('bc-stats')
+#' )
+#' }
+
+bcdc_list_organization_records <- function(organization) {
+  if(!has_internet()) stop("No access to internet", call. = FALSE) # nocov
+
+  cli <- bcdc_catalogue_client("action/organization_show")
+
+  r <- cli$get(query = list(id = organization, include_datasets = 'true'))
+  r$raise_for_status()
```

Comment:
@stephhazlitt you can catch statuses here and act on what is returned here using `r$status_code` I am _pretty_ sure.

Reply:
Maybe for these we catch the 404 with `r$status` code and the error message could say something like `"404: URL not found - you may have specified an invalid organization"`?

Reply:
I think that is definitely "good enough". We can revisit if we find a growing number of cases where users are getting a `404` with these functions (with valid names) where the 404 is from other internet issues. My bet is these will be very rare experiences?

Reply:
Attempt complete.